src/main/java/thrift/logic/commands/TagCommand.java

```diff
@@ -85,8 +85,9 @@ public CommandResult execute(Model model) throws CommandException {
                 ? ""
                 : String.format(MESSAGE_TAG_EXISTED, existedTags.toString());
 
-        actualIndex = model.getIndexInFullTransactionList(transactionToTag).get();
-        model.setTransactionWithIndex(index, updatedTransaction);
+        actualIndex = model.getIndexInFullTransactionList(transactionToTag).orElse(index);
+        model.setTransactionWithIndex(actualIndex, updatedTransaction);
+        model.updateFilteredTransactionList(Model.PREDICATE_SHOW_ALL_TRANSACTIONS);
```

Comment:
This line is not necessary as we will preserve the current filtered list last updated by the user

Reply:
Removed the requested line.